include/up-cpp/datamodel/builder/Payload.h

```diff
@@ -70,15 +78,22 @@ struct Payload {
 	///          will compile out.
 	/// @param data Data to be serialized and stored.
 	template <typename Serializer, typename ValueT>
-	Payload(Serializer s, const ValueT& data);
+	Payload(Serializer s, const ValueT& data) {
+		auto serializedData = Serializer::serialize(data);
+		//TODO: Payload format needs to be passed as argument or some other way.
+		//TODO: Following line of code is temporary and needs to be removed.
+		auto format = v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+		payload_ = std::make_tuple(serializedData, format);
+	}
 
 	/// @brief Creates a Payload builder with a provided pre-serialized data.
 	///
 	/// @param value_bytes A byte array containing the serialized payload.
 	/// @param format The data format of the payload in value_bytes.
 	///
 	/// @throws std::out_of_range If format is not valid for v1::UPayloadFormat
-	Payload(const std::vector<uint8_t>& value_bytes, v1::UPayloadFormat format);
+	Payload(const std::vector<uint8_t>& value_bytes,
+	        const v1::UPayloadFormat& format);
```

Comment:
Thanks for pointing this out. I have also read several articles where they have mentioned "In C++, passing small objects like integers, characters, booleans, and enumerations by value is usually more efficient than passing them by reference. This is because passing by reference involves the overhead of dereferencing a pointer, which can be more costly than copying a small object. The `v1::UPayloadFormat` is an enumeration, so it's a small object. Therefore, it would be more efficient to pass it by value rather than by reference."
I have kept the const keyword as it is but removed the reference now

Reply:
Learned new thing today and I will keep this in mind going forward